FlowCal/excel_ui.py

```diff
@@ -357,57 +360,78 @@ def process_beads_table(beads_table,
-                       else 'linear',
+                xscale='log' if sc_amp_type[0][0] else 'logicle',
+                yscale='log' if sc_amp_type[1][0] else 'logicle',
                 full_output=True)
 
             # Plot forward/side scatter density plot and fluorescence histograms
             if plot:
                 if verbose:
                     print("Plotting density plot and histogram...")
-                # Define density plot parameters
+                # Density plot parameters
                 density_params = {}
                 density_params['mode'] = 'scatter'
-                if beads_sample_gated.amplification_type(sc_channels[0])[0]:
-                    density_params['xscale'] = 'log'
-                else:
-                    density_params['xscale'] = 'linear'
-                if beads_sample_gated.amplification_type(sc_channels[1])[0]:
-                    density_params['yscale'] = 'log'
-                else:
-                    density_params['yscale'] = 'linear'
                 density_params["title"] = "{} ({:.1f}% retained)".format(
                     beads_id,
                     beads_sample_gated.shape[0] * 100. / beads_sample.shape[0])
+                # Axis scaling will be logarithmic if data was acquired with a
+                # log amplifier, and logicle otherwise.
+                sc_amp_type = beads_sample_gated.amplification_type(sc_channels)
+                density_params['xscale'] = \
+                    'log' if sc_amp_type[0][0] else 'logicle'
+                density_params['yscale'] = \
+                    'log' if sc_amp_type[1][0] else 'logicle'
+                # Beads have a tight distribution, so axis limits will be set
+                # from 0.75 decades below the 5th percentile to 0.75 decades
+                # above the 95th percentile.
+                density_params['xlim'] = \
+                    (np.percentile(beads_sample_gated[:, sc_channels[0]],
+                                   5) / (10**0.75),
+                     np.percentile(beads_sample_gated[:, sc_channels[0]],
+                                   95) * (10**0.75),
+                     )
+                density_params['ylim'] = \
+                    (np.percentile(beads_sample_gated[:, sc_channels[1]],
+                                   5) / (10**0.75),
+                     np.percentile(beads_sample_gated[:, sc_channels[1]],
+                                   95) * (10**0.75),
+                     )
+                # Histogram plot parameters
+                # Axis scaling will be logarithmic if data was acquired with a
+                # log amplifier, and logicle otherwise.
+                cluster_amp_type = \
+                    beads_sample_gated.amplification_type(cluster_channels)
+                hist_params = \
+                    [{'xscale': 'log' if ch_amp_type[0] else 'logicle'}
+                     for ch_amp_type in cluster_amp_type]
```

Comment:
same as mentioned elsewhere: I think amplifier scale and plotting (and gating) scale are orthogonal, so I would still go with logicle gating, even with a log amplifier.